overlord/configstate/applyconfighandler.go

```diff
@@ -38,11 +35,22 @@ func (h *applyConfigHandler) Before() error {
 // Done is called by the HookManager after the apply-config hook has exited
 // successfully.
 func (h *applyConfigHandler) Done() error {
+	// Save any configurations changes the hook may have made
+	h.transaction.Commit()
```

Comment:
Done. I've added `Cache/Cached` logic into the `Context`, and also added a deferring mechanism there which is now utilized by the `snapctl set` command.